src/server/scripts/EasternKingdoms/Stratholme/boss_baroness_anastari.cpp

```diff
@@ -32,50 +32,59 @@ enum Spells
 {
     SPELL_BANSHEEWAIL       = 16565,
     SPELL_BANSHEECURSE      = 16867,
-    SPELL_SILENCE           = 18327
-    //SPELL_POSSESS           = 17244
+    SPELL_SILENCE           = 18327,
+    SPELL_POSSESS           = 17244,
+    SPELL_POSSESSED         = 17246
+};
+
+enum BaronessAnastariEvents
+{
+    EVENT_SPELL_BANSHEEWAIL     = 1,
+    EVENT_SPELL_BANSHEECURSE    = 2,
+    EVENT_SPELL_SILENCE         = 3,
+    EVENT_SPELL_POSSESS         = 4,
+    EVENT_INVISIBLE             = 5
 };
 
 class boss_baroness_anastari : public CreatureScript
 {
 public:
     boss_baroness_anastari() : CreatureScript("boss_baroness_anastari") { }
 
-    CreatureAI* GetAI(Creature* creature) const override
-    {
-        return GetStratholmeAI<boss_baroness_anastariAI>(creature);
-    }
-
     struct boss_baroness_anastariAI : public ScriptedAI
     {
         boss_baroness_anastariAI(Creature* creature) : ScriptedAI(creature)
         {
-            Initialize();
             instance = me->GetInstanceScript();
         }
 
-        void Initialize()
-        {
-            BansheeWail_Timer = 1000;
-            BansheeCurse_Timer = 11000;
-            Silence_Timer = 13000;
-            //Possess_Timer = 35000;
-        }
-
         InstanceScript* instance;
-
-        uint32 BansheeWail_Timer;
-        uint32 BansheeCurse_Timer;
-        uint32 Silence_Timer;
-        //uint32 Possess_Timer;
+        
+        bool invisible;
+        Unit* possessedTarget;
+        uint32 invisibleTimer; // after 2 minutes dispell possessed
 
         void Reset() override
         {
-            Initialize();
+            if (possessedTarget)
+            {
+                if (possessedTarget->HasAura(SPELL_POSSESSED) && possessedTarget->IsAlive())
+                {
+                    possessedTarget->RemoveAurasDueToSpell(SPELL_POSSESSED);
+                    possessedTarget->RemoveAurasDueToSpell(SPELL_POSSESS);
+                    possessedTarget->SetObjectScale(1.0f);
```

Comment:
what happens if the target is dead with object scale set to 1.5f ?

Reply:
wouldn't it just scale back? or am i missing something

Reply:
It does scale back. The reset is only to make sure when the party wipes if someone is still possessed then dispells him

Reply:
I think the scaling itself is handled by a spell instead. This also overrides auras that increase/change the scale of a unit.

Reply:
Nothing on sniffs show scaling spells